Fixes for running with open boundary conditions over land

fjansson opened the pull request:
*land surface model*: fix sign of rain flux, add some limits and diagnostics.
*microphysics*: run only up to kmax. If microphysics is done at k1, and open boundaries keep the k1 layer raining, it can form an "infinite" rain source, since cloud water is not removed from k1.
*modgenstat*: don't use `halflev_kappa` with open boundaries, because it can cause out-of bounds accesses at the top when w != 0. This change only affects statistics, not the time stepping.

fjansson committed:
modbulkmicro: run microphysics only up to kmax

fjansson committed:
modlsm: fix sign of rain rate and cap soil moisture

fjansson committed:
modgenstat: with openBC, don't use halflev_kappa to interpolate (because broken)

fjansson committed:
OpenBC: fill all ghost cells on an open boundary, not just one.

For using higher-order advection schemes which need more neighbor points.
The ghost cell planes further away are filled with the same values as
the origininal single plane.

fjansson committed:
call abort instead of stop in netcdf error checking to get backtrace

fjansson committed:
modforces: fix sx,sy for OpenBC in Coriolis routine

fjansson committed:
advection: add missing sy in 2nd order advecv

found by Frans Liqui Lung

fjansson committed:
timestep: update also ghost cells for u,v,w for open boundary conditions
found by Frans Liqui Lung
With open boundaries, u,v,w need some boundary cells to be time-stepped:
u0(i2,j,k), v0(i,j2,k) and w0(i,j,k1)

esclapez commented on src/tstep.f90:
```diff
@@ -179,8 +179,8 @@ subroutine tstep_update
   do k = 1, k1
-    do j = 2, j1
-      do i = 1, i1
+    do j = 2, j2     ! i2, j2 here to include one ghost cell,
```
Throughout both `tstep_****` routines, `j` starts at 2 while `i` starts at 1. Not sure it matters, but switching to using 1 on both might be better.

fjansson replied:
I agree, let's handle both directions in the same way. My feeling is that starting at 2 is sufficient. @CasparJungbacker, @franslql ?

fjansson replied:
I made the i,j loops start at 2. The loops that need to end at i2,j2 for open boundaries are now only extended to i2,j2 when needed: if open boundaries are enabled and the MPI tile is at the far domain border. Frans' open boundaries test case over sea still runs.

fjansson committed:
timestep and OpenBC: loop start and end point adjutment

start loops from i=2, not 1. starting point adjusted for consistency
with j and a bit less work, otherwise no practical difference.

loop end point extended to j2 only when open boundaries are enabled and
only on the MPI tiles where it's needed.
